WebHostLib/templates/playerOptions/macros.html

```diff
@@ -0,0 +1,264 @@
+                    <div class="option-divider">&nbsp;</div>
+                {% endif %}
+                {% for location_name in world.location_names|sort %}
+                    <div class="option-entry">
+                        <input type="checkbox" id="{{ option_name }}-{{ location_name }}" name="{{ option_name }}" value="{{ location_name }}" {{ "checked" if location_name in option.default }} />
+                        <label for="{{ option_name }}-{{ location_name }}">{{ location_name }}</label>
+                    </div>
+                {% endfor %}
+            </div>
+        </td>
+    </tr>
+{% endmacro %}
+
+{% macro ItemSet(option_name, option, world) %}
+    <tr>
+        <td>{{ OptionTitle(option_name, option) }}</td>
+        <td>
+            <div class="option-container">
+                {% for group_name in world.item_name_groups.keys()|sort %}
+                    {% if group_name != "Everything" %}
+                        <div class="option-entry">
+                            <input type="checkbox" id="{{ option_name }}-{{ group_name }}" name="{{ option_name }}" value="{{ group_name }}" {{ "checked" if group_name in option.default }} />
+                            <label for="{{ option_name }}-{{ group_name }}">{{ group_name }}</label>
+                        </div>
+                    {% endif %}
+                {% endfor %}
+                {% if world.item_name_groups.keys()|length > 1 %}
+                    <div class="option-divider">&nbsp;</div>
+                {% endif %}
+                {% for item_name in world.item_names|sort %}
+                    <div class="option-entry">
+                        <input type="checkbox" id="{{ option_name }}-{{ item_name }}" name="{{ option_name }}" value="{{ item_name }}" {{ "checked" if item_name in option.default }} />
+                        <label for="{{ option_name }}-{{ item_name }}">{{ item_name }}</label>
+                    </div>
+                {% endfor %}
+            </div>
+        </td>
+    </tr>
+{% endmacro %}
+
+{% macro OptionSet(option_name, option) %}
+    <tr>
+        <td>{{ OptionTitle(option_name, option) }}</td>
+        <td>
+            <div class="option-container">
+                {% for key in option.valid_keys %}
+                    <div class="option-entry">
+                        <input type="checkbox" id="{{ option_name }}-{{ key }}" name="{{ option_name }}" value="{{ key }}" {{ "checked" if key in option.default }} />
+                        <label for="{{ option_name }}-{{ key }}">{{ key }}</label>
+                    </div>
+                {% endfor %}
+            </div>
+        </td>
+    </tr>
+{% endmacro %}
+
+{% macro OptionTitle(option_name, option) %}
+    <label for="{{ option_name }}">
+        {{ option.display_name|default(option_name) }}:
+        <span class="interactive" data-tooltip="{% filter dedent %}{{(option.__doc__ | default("Please document me!"))|escape }}{% endfilter %}">(?)</span>
```

Comment:
One common way to write doc strings is with the triple quotes on their own line.
```python
"""
documentation text here
"""
```
That newline at the beginning is showing up in a bunch of the tooltips.

```suggestion
        <span class="interactive" data-tooltip="{% filter dedent %}{{(option.__doc__.strip('\n') | default("Please document me!"))|escape }}{% endfilter %}">(?)</span>
```

We don't want to strip all whitespace with `.strip()` because that would mess up the `dedent`

This just removes newlines at the beginning and the end.

Reply:
https://github.com/ArchipelagoMW/Archipelago/pull/3262

Reply:
Stripping the text before the dedent might mess up the filter, but stripping it after the dedent lets us do the necessary touch-ups.
```py
@app.template_filter("dedent")
def filter_dedent(text: str):
    return dedent(text).strip("\n ")
```

Reply:
That doesn't solve the problem with the 2nd style of doc string addressed in this: https://github.com/ArchipelagoMW/Archipelago/pull/3262

Reply:
That solution also supports someone who does want an indented first line, with something like this:
```python
class MyOption(Option):
    """
       My neatly formatted paragraph
    is so beautiful.
    """"
```